src/ISA/ISA/ARCTypes/CompositeCell.fs

```diff
@@ -1,6 +1,67 @@
 ﻿namespace ISA
 
+// TODO: The design principles say we should use Erase for js compilation. This would also 
+
+//[<Erase>]
+//[<AttachMembersAttribute>]
+// FABLE: Erased unions with multiple cases cannot have more than one field: Test.CompositeCell
+//type CompositeCell = 
+//    | Term of string
+//    | Freetext of {| t: string; f: float |}
+//    | WithUnit of {| t: string; b: bool |}
+
+//    member this.getString =
+//        match this with
+//        | Term s -> s
+//        | Freetext t -> t.t
+//        | WithUnit t -> t.t
+
+//let test_t = Term ("Test me")
+//let test_f = Freetext {|t = "Test me"; f = 12.|}
+
+//test_t.getString |> printfn "%A" // Does return "undefined"
+
+// if we erase we cannot use members
+
 type CompositeCell = 
     | Term of OntologyAnnotation
-    | Freetext of string
-    | WithUnit of string*OntologyAnnotation
\ No newline at end of file
+    | FreeText of string
+    | Unit of string*OntologyAnnotation
+
+    member this.isUnit = match this with | Unit _ -> true | _ -> false
+    member this.isTerm = match this with | Term _ -> true | _ -> false
+    member this.isFreetext = match this with | FreeText _ -> true | _ -> false
+
+    member this.toUnitCell() =
```

Comment:
These "to" functions should be well documented as intuition about what exactly they do might vary.